test/fixtures/cookbooks/test/recipes/replace_or_add.rb

```diff
@@ -22,6 +22,18 @@
 
 replace_or_add 'Operation 4' do
   path '/tmp/dangerfile2'
-  pattern 'ssh-rsa AAAAB3NzaC1yc2EAAAADDEADBEEF.*'
+  pattern 'ssh-dsa AAAAB3NzaC1yc2EAAAADDEADBEEF.*'
```

Comment:
The test replaces a matching line with a nil line.  It is easier to test for and count a unique pattern.  We do lots of other tests with the ssh-rsa lines. This ssh-dsa line is here for this test only.